Output File Refactors + AWS Scripts

sstevenshang opened the pull request:
This in PR:
1. Refactored arg parser of SPMAEL module.
2. Added various helper functions to help write/delete files from HDFS (adapted to AWS S3 as well).
3. Added scripts to automate AWS operations (see `aws/deploy_to_s3.sh` and `aws/run_in_emr.sh`).

This should close the following issues:
- https://github.com/ncsa/Genomics_EpiQuant/issues/8
- https://github.com/ncsa/Genomics_EpiQuant/issues/12

Note:
- The new file `aws/genomics-epiquant-beta-keypair.pem` is the key-pair file used for SSH-ing into the EMR cluster. Keep this here in this repo for now. But should be deleted in the future for best practice.

sstevenshang committed:
Added function to write output directly to fs

sstevenshang committed:
Refactored arg parser for SPAEML

sstevenshang committed:
Merge branch 'output_file_fix' into aws_dev

sstevenshang committed:
Reverted to use saveAsTextFile until we figure out the best way to output files

sstevenshang committed:
Fixed existing output directory on local

sstevenshang committed:
Fixed a bug that causes failure on aws

sstevenshang committed:
Output is now written to single files, tested on both local and AWS

sstevenshang committed:
Added comments, deleted unused function

sstevenshang committed:
Added script for deploying to s3

sstevenshang committed:
Added script for running EMR cluster

sstevenshang committed:
Updated REAME

jacobrh91 committed:
Corrected small typo

jacobrh91 reviewed (commented):
These changes are good. A few things:

Lets not upload the keypair.pem file on GitHub, as that is probably insecure. We need to just distribute the key to each other privately instead.

In StepwiseModelSelection.scala:
- Lets set the default sparkMaster to just local[] instead of local[3]

The refactoring of the input parameters and writing to HDFS looks great.

Also, to fix ISSUE #8, instead of automatically removing the output directory if it already exists, lets instead throw an error and force the user to either remove it manually, or put the current output in another location. For real world data sets, this will run for a long time, and we don't want the output to be deleted or rewritten so easily.

Does this still work on a non-Hadoop filesystem (your local machine for example)? If you've tested and confirmed that it still does, then we can merge this thing

sstevenshang committed:
Deleted keypair.pem

sstevenshang committed:
Changed default master to local

sstevenshang committed:
Throw error when output dir exists instead of deleting it

sstevenshang committed:
Accidentally added output file to git

sstevenshang commented:
Sounds good. I made some updates:
- Deleted the key pair.
- Changed default sparkMaster to "local". 
- Changed it to throw error when output directory exists.

I've tested and verified that this works on local machines. Should be ready to merge
